### PR TITLE
Fix some redundant checks for non-NULL values raised up by clang

### PR DESCRIPTION
**Pull request by Kakadu**
Some warnings raised during compiling with clang for stm32f4cube. In `vtparse.c ` we compare unsigned value with -1. In another file we test that the pointer to global value is not null.

@anton-bondarev 

P.S. It is a part of #1023 for now.

````
src/drivers/char/terminal/vtparse.c:49:24: error: comparison of constant -1 with expression of type 'char' is always false
      [-Werror,-Wtautological-constant-out-of-range-compare]
                if (token->attrs_len == -1) {
                    ~~~~~~~~~~~~~~~~ ^  ~~

````

### CODE REVIEW

File: src/drivers/char/terminal/vtparse.c [hunk 1]
```diff
@@ -46,10 +46,11 @@ static void do_action(struct vtparse *parser, vt_action_t action, char ch) {
 		break;
 
 	case VT_ACTION_COLLECT:
-		if (token->attrs_len == -1) {
+		// Buffer overflow check may be needed
+		//if (token->attrs_len == -1) {
```
Comment (AntonKozlov) [inline src/drivers/char/terminal/vtparse.c]: As far as I understood, attr overflow checked in other states (attrs buffer is VTPARSE_TOKEN_PARAMS_MAX (2) long). If it overflows, then len is assigned to `-1` and this as guard conditions to all states.
The problem seems to be that `attrs_len` is of `char` type, which 
a) could be signed or unsigned
b) is not what you'd expect for length type.

I'd suggest to revert this code and change attrs_len to `int` (probably reorganized vt_token structure to  get rid of unnecesary member padding)

@abusalimov could have something to say (as the author)

File: src/net/util/show_packet.h [hunk 2]
```diff
@@ -20,7 +20,7 @@ static inline void show_packet(uint8_t *raw, int size, char *title) {
 	if (!&mod_logger)
 		return;
 
-	if (&mod_logger.logging &&
+	if (//&mod_logger.logging &&
```
Comment (AntonKozlov) [inline src/net/util/show_packet.h]: This seems as inlined version of `logging_raw` function, which actually checks for this address. I'd suggest to totally remove commented line and replace also runtime check for `logging` in `logging_raw` with assert

We could also ask author (@vloginova) for better way

File: src/drivers/char/terminal/vtparse.c (hunk 1 above)
Comment (Kakadu) [inline src/drivers/char/terminal/vtparse.c]: @AntonKozlov I'm terribly confused by this. I thout that attr_len is unsigned but it seems that it have type `char`. But you can see how clang raises a warning above (I just added this).

Comment (anton-bondarev) [inline src/drivers/char/terminal/vtparse.c]: @Kakadu May be in some platforms char can be unsigned? I offer to change type of **token->attrs_len** to **int** and remove comments

File: src/net/util/show_packet.h (hunk 2 above)
Comment (AntonKozlov) [inline src/net/util/show_packet.h]: Ok, let's leave this commented, but with comment why it's commented out

Comment (abusalimov) [inline src/net/util/show_packet.h]: Checking for `&whatever` to be non-null is indeed redundant, I guess, this should be removed. AFAICS, this file, and these lines, in particular, were added by @Deryugin, Denis, could you elaborate on this, please?

Comment (AntonKozlov) [inline src/net/util/show_packet.h]: Please consider also `logging_raw` function, which called like
```
#define log_logp(level, fmt, ...) \
        if (&mod_logger) \
                logging_raw(&mod_logger.logging, level, "%s: %s: " fmt "\n", \
```
and defined to be 
```
void logging_raw(struct logging *logging, int level, const char* fmt, ...) {
        if (logging
```

I find this definition and the redundant check at least consistent. 
In case the check deleted `logging_raw` should be altered similar way, I suppose

**Comment by anton-bondarev** [comment]
Merged by #1029
